### PR TITLE
pd: send http header on connect.

### PR DESCRIPTION
**Pull request by huachaohuang**
For [pd #241](https://github.com/pingcap/pd/pull/241).

### CODE REVIEW

File: src/pd/client.rs [hunk 1]
```diff
@@ -55,6 +58,51 @@ fn send_msg(stream: &mut TcpStream, msg_id: u64, message: &Request) -> Result<(u
...
+    let hosts = try!(parse_urls(addr));
+
+    for host in &hosts {
+        let mut stream = match make_std_tcp_conn(host.as_str()) {
+            Ok(stream) => stream,
+            Err(_) => continue,
+        };
+
+        // Send a HTTP header to tell PD to hijack this connection for RPC.
+        let header = b"GET /pd/rpc HTTP/1.0\r\n\r\n";
```
Comment (siddontang) [inline src/pd/client.rs]: define a const val

File: src/pd/client.rs [hunk 2]
```diff
@@ -55,6 +58,51 @@ fn send_msg(stream: &mut TcpStream, msg_id: u64, message: &Request) -> Result<(u
...
+        try!(stream.set_write_timeout(Some(Duration::from_secs(SOCKET_WRITE_TIMEOUT))));
+        match stream.write(header) {
```
Comment (siddontang) [inline src/pd/client.rs]: why not using try! here?

**Comment by siddontang** [comment]
Rest LGTM

**Comment by huachaohuang** [comment]
I need to merge this with [pd #241](https://github.com/pingcap/pd/pull/241).
@BusyJay @disksing @zhangjinpeng1987 PTAL.

File: src/pd/client.rs [hunk 3]
```diff
@@ -55,6 +58,51 @@ fn send_msg(stream: &mut TcpStream, msg_id: u64, message: &Request) -> Result<(u
     Ok((id, resp.take_pd_resp()))
 }
 
+fn parse_urls(addr: &str) -> Result<Vec<String>> {
+    let mut hosts = Vec::new();
+    for a in addr.split(',') {
+        let url = match Url::parse(a) {
```
Comment (BusyJay) [inline src/pd/client.rs]: Use `box_try!` instead.

File: src/pd/client.rs [hunk 4]
```diff
@@ -55,6 +58,51 @@ fn send_msg(stream: &mut TcpStream, msg_id: u64, message: &Request) -> Result<(u
...
+            Ok(url) => url,
+            Err(e) => return Err(box_err!("parse url failed: {:?}", e)),
+        };
+        if url.host_str().is_none() || url.port().is_none() {
```
Comment (BusyJay) [inline src/pd/client.rs]: If the url is `http://example.com/`, will `url.port()` returns `None`?

**Comment by huachaohuang** [comment]
@BusyJay @disksing PTAL

**Comment by ngaut** [comment]
LGTM

**Pull request merged**